qiskit/qasm3/__init__.py

```diff
@@ -79,6 +79,21 @@
     qasm_string = qasm3.dumps(qc, experimental=qasm3.ExperimentalFeatures.SWITCH_CASE_V1)
 
 
+.. note::
+
+    All features enabled by the experimental flags are naturally transient.  If it becomes necessary
+    to remove flags, they will be subject to `the standard Qiskit deprecation policy
+    <https://qiskit.org/documentation/deprecation_policy.html>`__.  We will leave these experimental
+    flags in place for as long as is reasonable.
```

Comment:
I'm wondering what our default export story is in the case of the spec changing with the experimental flag. For example, if switch gets adopted with a new incompatible syntax. Is the plan for the next terra release is to make the output default to the new switch syntax? If so then if the flag is set it gets the same version we're exporting now?

Not that I think we need to document that level of detail here, but I'm just trying to figure out what our development plan/policy will be around these experimental flags.

Reply:
My intent was that the flag causes export with the semantics defined by one specific commit - the text I put in the docs for the `SWITCH_CASE_V1` flag was

> Support exporting switch-case statements as proposed by openqasm/openqasm#463 at commit openqasm/openqasm@bfa787aa3078.

So in your situation, if switch-case gets merged with incompatible syntax, then my intention was that Terra's default should be to export switch with the merged syntax, but if the flag is passed, then export with the old syntax as defined at the linked commit.

Reply:
Ok, that sounds good. That's what I expected I just wanted to make sure we were on the same page if this comes up in the future.